packages/parameters/src/AppConfigProvider.ts

```diff
@@ -0,0 +1,115 @@
+import { BaseProvider, DEFAULT_PROVIDERS } from './BaseProvider';
+import {
+  AppConfigDataClient,
+  StartConfigurationSessionCommand,
+  GetLatestConfigurationCommand,
+} from '@aws-sdk/client-appconfigdata';
+import type {
+  StartConfigurationSessionCommandInput,
+  GetLatestConfigurationCommandInput,
+} from '@aws-sdk/client-appconfigdata';
+import type { AppConfigGetOptionsInterface } from './types/AppConfigProvider';
+
+class AppConfigProvider extends BaseProvider {
+  public client: AppConfigDataClient;
+  private application: string;
+  private environment: string;
+  private latestConfiguration: Uint8Array | undefined;
+  private token: string | undefined;
+
+  /**
+   * It initializes the AppConfigProvider class'.
+   * *
+   * @param {AppConfigGetOptionsInterface} config
+   */
+  public constructor(options: AppConfigGetOptionsInterface) {
```

Comment:
I think the `options` parameter should have its own interface, similar to what [done here for the DynamoDBProvider](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/5144b3b974bc1f988c2d3d0599f6f05a76cc315c/packages/parameters/src/types/DynamoDBProvider.ts#L4-L10).

This interface would look like this:
```ts
interface AppConfigProviderOptions {
  application: string
  environment: string
  clientConfig?: AppConfigDataClientConfig
}
```

This is in line with the Python's implementation, which requires users to specify an `application` and `environment` (both mandatory) and allows for an optional config object that will be passed to the SDK client. This config object will allow users to set things like the AWS Region or pass explicit AWS credentials.

Reply:
As an aside, while writing this comment I realized that I will have to adjust the `SSMProvider` to have a similar shape.

Reply:
> There's then a third optional config object that will be passed to the SDK client (note the `AppConfigDataClientConfig` type). This config object will allow users to set things like the AWS Region or pass explicit AWS credentials.

If we extract config from options, how to pass it to the new instance of the class? I asked you about that in [SecretsProvider PR](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1206/files#diff-ba2b2c24f5637430ac1de6cf8f6de4832ee5c12149e9563de13523ea38c904ecR7).

Reply:
I don't see any comment at the line you have highlighted

Reply:
But aside from the comment I might have missed (sorry if I did):
- `AppConfigProviderOptions` is the interface used for the parameter passed to the Class constructor of a provider.
- `AppConfigGetOptionsInterface` it's the interface used for the `options` parameter of the `get`/`_get` methods

The former has an optional `clientConfig` parameter that allows the user to configure the behavior of the entire AWS SDK client that is instantiated upon class instantiation. The configs passed here will apply to all the API calls made with this client.

The latter instead has an optional `sdkOptions` parameter that allows the user to configure the behavior of a specific API call (made via the SDK).

Reply:
> I don't see any comment at the line you have highlighted

My bad, I didn't submit the review. The same for the `SSMProvider` you asked me to review and which I did. I'm not really experienced in reviewing. I think I did it many times and thought why you are not responding to my comments 🤣

Reply:
Oh I see! 

I think while you're in review mode you have two choices:
- "Post as single comment" (or named something similar) which basically adds the comment immediately and as standalone
- "Add as review" - In this case it'll be marked as "pending" and it won't show up until you actually submit the review (green button top-right of the review page)

Maybe you were doing the second 😁